addons/html_builder/static/src/builder/snippets_menu.js

```diff
@@ -210,9 +212,28 @@ export class SnippetsMenu extends Component {
 
     onBeforeUnload(event) {
         if (!this.isSaving && this.editor.shared.dirty.isEditableDirty()) {
+            event.preventDefault();
```

Comment:
Just to be compliant with [the doc](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event)